src/theme/colour-primitive/theme-helper.ts

```diff
@@ -28,18 +28,61 @@ export const ColourSpec: ThemeCollectionSpec<
     defaultValue: "lifesg",
 };
 
+// Dark colour spec - uses explicit dark sets where available, falls back to light sets
+const DarkColourSpec: ThemeCollectionSpec<ColourCollectionsMap, ColourScheme> =
+    {
+        collections: {
+            // Currently all schemes use light primitive colours as fallback
+            // In the future, add custom dark primitive sets like:
+            // lifesg: LifeSgDarkColourSet,
+            lifesg: LifeSgColourSet,
+            bookingsg: BookingSgColourSet,
+            rbs: RBSColourSet,
+            mylegacy: MyLegacyColourSet,
+            ccube: CCubeColourSet,
+            oneservice: OneServiceColourSet,
+            pa: PAColourSet,
+            a11yplayground: A11yPlaygroundColourSet,
+        },
+        defaultValue: "lifesg",
+    };
+
 export const getPrimitiveColour = (key: keyof PrimitiveColourSet) => {
     return (props: StyledComponentProps): string => {
         const theme = props.theme;
-        const colorSet: PrimitiveColourSet = getCollection(
-            ColourSpec,
+        const isDarkMode = theme?.colourMode === "dark";
+
+        // Select the appropriate colour spec based on theme mode
+        const spec = isDarkMode ? DarkColourSpec : ColourSpec;
+        const colourSet: PrimitiveColourSet = getCollection(
+            spec,
             theme?.colourScheme
         );
 
-        if (theme?.overrides?.primitiveColour) {
-            return getValue(colorSet, key, theme.overrides.primitiveColour);
+        // Handle case where colourSet might be undefined
+        if (!colourSet) {
```

Comment:
on second thought, is there a case where `colourSet` is undefined? as we would always fallback to `ColourSpec` and the `lifesg` scheme if the theme is not available